saleor/core/db/expressions.py

```diff
@@ -0,0 +1,49 @@
+from django.db.models import JSONField
+from django.db.models.expressions import Expression
+
+
+class JsonCat(Expression):
+    # It's possible to recursevily create that expression to allow to pass
+    # more than 2 arguments. If needed it can be extended in future
+    template = "%(left)s || %(right)s"
+    output_field = JSONField()
+
+    def __init__(self, left_arg, right_arg):
+        super().__init__(output_field=self.output_field)
+        self.__validate_argument(left_arg)
+        self.__validate_argument(right_arg)
+        self.left = left_arg
+        self.right = right_arg
+
+    def __validate_argument(self, arg):
+        if not hasattr(arg, "resolve_expression"):
+            # make sure that argument is Expression
+            raise TypeError("%r is not an Expression", arg)
+        if hasattr(arg, "output_field") and not isinstance(arg.output_field, JSONField):
```

Comment:
I found out the easiest way to for implementing whole idea to force usage of `Value` . 
Maybe there is better way to do it?